File: kotlin/src/romeo/C0018_RANDOM.kt
package romeo

import java.security.SecureRandom
import java.util.*


class C0018_RANDOM {
    fun bad() {
        val rand = Random()
        /* FLAW: seed is static, making the numbers always occur in the same sequence */
        rand.setSeed(123456)
        Util.writeLine("Random int: " + rand.nextInt(100))
    }

    fun good() {
        /* FIX: use SecureRandom to be cryptographically secure */
        val rand = SecureRandom()
        Util.writeLine("Random int: " + rand.nextInt(100))
    }
}

File: swift/src/main/swift/C0018_RANDOM/C0018_RANDOM.swift
package swift.C001B_PERSCOOKIE.swift

import SQLite3

func bad() {
  var n: Int = 3
  // Flaw
  sqlite3_randomness(10, &n)
}

func good() {
  let randomInt = Int.random(in: 1..<5)
}
